### ISSUE TITLE
Incorrect heap usage statistics

### ISSUE THREAD

**Issue by splitice** [node-ffi-napi/node-ffi-napi#80]
When using this addon (version 3.0.0) heap usage information is incorrect.

![](https://drive.google.com/uc?id=1bE-jdJMSO4Hg2gk2yVvEkllt3bYhpG2L)

![image](https://user-images.githubusercontent.com/468579/83213850-7eaafe80-a1a6-11ea-841f-c541bfe5329f.png)

This is being used in my node-fallocate fork https://github.com/HalleyAssist/node-fallocate

**Comment by addaleax** [comment]
Yeah, this is unfortunate, but Node 14 required some internals changes that forced node-ffi-napi to start intentionally creating Buffers that are too large. I don’t think there’s any way around this that doesn’t break node-ffi-napi users. :/

**Comment by splitice** [comment]
@addaleax that sucks. Although I can't say I'm surprised. I've fixed a few addons myself due that change.

Is there a corresponding v8 / node issue for that the internal change creating incompatibilities like this?

Perhaps the buffers could be hidden? I'd rather have some additional untracked memory than having all the valuable information hidden.

**Comment by addaleax** [comment]
@splitice Yes, it does kind of suck.

https://bugs.chromium.org/p/v8/issues/detail?id=9908 is the V8 feedback issue for this particular issue. 

More concretely, the problem is that `node-ffi-napi` and `ref-napi` provide a way of creating a Buffer from the same underlying pointer as another Buffer, with a specified size – the V8 API just plainly doesn’t support that, so the workaround is to create a maximum-size Buffer and then always return slices of that. That works out most of the time. The actual memory usage isn’t changed, though.

I don’t know any way in which the memory usage can be reported differently here either, sorry.
